pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/ConfigBeanRegistryImpl.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.configdriven.runtime;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.helidon.common.LazyValue;
+import io.helidon.common.config.Config;
+import io.helidon.common.config.ConfigException;
+import io.helidon.pico.api.Bootstrap;
+import io.helidon.pico.api.PicoException;
+import io.helidon.pico.api.PicoServiceProviderException;
+import io.helidon.pico.api.PicoServices;
+import io.helidon.pico.api.PicoServicesHolder;
+import io.helidon.pico.api.Qualifier;
+import io.helidon.pico.api.Resettable;
+import io.helidon.pico.configdriven.api.ConfigBeanFactory;
+import io.helidon.pico.configdriven.api.NamedInstance;
+
+/**
+ * The default implementation for {@link ConfigBeanRegistry}.
+ */
+@SuppressWarnings("unchecked")
+class ConfigBeanRegistryImpl implements ConfigBeanRegistry, Resettable {
```

Comment:
lost history on this also

Reply:
Used to be `DefaultPicoConfigBeanRegistry`, git seems not to understand that. Not sure why